plugins/modules/fabric_devices_workflow_manager.py

```diff
@@ -0,0 +1,4080 @@
+            "ip_l3_handoff_details": {}
+        })
+        delete_ip_l3_handoff = []
+        non_existing_ip_l3_handoff = []
+
+        # Check which IP L3 Handoff exists and which doesnot
+        for item in have_ip_l3_handoff:
+            if not item:
+                non_existing_ip_l3_handoff.append(item)
+            else:
+                delete_ip_l3_handoff.append(item)
+
+        for item in delete_ip_l3_handoff:
+            id = item.get("id")
+            try:
+                payload = {"id": id}
+                task_name = "delete_fabric_device_layer3_handoff_with_ip_transit_by_id"
+                task_id = self.get_taskid_post_api_call("sda", task_name, payload)
+                if not task_id:
+                    self.msg = (
+                        "Unable to retrive the task_id for the task '{task_name}'."
+                        .format(task_name=task_name)
+                    )
+                    self.set_operation_result("failed", False, self.msg, "ERROR")
+                    return self
+
+                success_msg = (
+                    "Successfully deleted the fabric device IP Transit L3 Handoff with id '{id}'."
+                    .format(id=id)
+                )
+                self.get_task_status_from_tasks_by_id(task_id, task_name, success_msg).check_return_status()
+            except Exception as msg:
+                self.msg = (
+                    "Exception occured while deleting the IP L3 Handoff in the fabric device with IP '{ip}': {msg}"
+                    .format(ip=device_ip, msg=msg)
+                )
+                self.log(self.msg, "ERROR")
+                self.status = "failed"
+                return self
+
+        result_fabric_device_response.get("ip_l3_handoff_details").update({
+            "Deleted L2 Handoff": delete_ip_l3_handoff,
+            "Non existing L2 Handoff": non_existing_ip_l3_handoff
+        })
+        if delete_ip_l3_handoff:
+            result_fabric_device_msg.update({
+                "ip_l3_handoff_details": "IP L3 Handoffs deleted successfully."
+            })
+        else:
+            result_fabric_device_msg.update({
+                "ip_l3_handoff_details": "IP L3 Handoffs are not found in the Cisco Catalyst Center."
+            })
+
+        self.msg = "Successfully deleted the IP L3 Handoffs."
+        self.status = "success"
+        return self
+
+    def delete_fabric_devices(self, fabric_devices):
+        """
+        Delete fabric devices, Border Settings, L2 Handoffs, L3 Handoff with SDA Transit,
+        L3 Handoff with IP Transit in Cisco Catalyst Center with fields provided in playbook.
+
+        Parameters:
+            fabric_devices (dict): SDA fabric devices under a fabric site playbook details.
+        Returns:
+            self (object): The current object with updated desired Fabric Devices information.
+        Description:
+
+        """
+
+        fabric_name = fabric_devices.get("fabric_name")
```

Comment:
if not fabric_name:
    self.log("Error: 'fabric_name' is missing from input.", "ERROR")
    self.set_operation_result("failed", False, "Fabric name is required.", "ERROR")
    return self